config/management/genesis/src/waypoint.rs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use executor::db_bootstrapper;
+use libra_global_constants::WAYPOINT;
+use libra_management::{
+    error::Error,
+    secure_backend::{SharedBackend, ValidatorBackend},
+    storage::StorageWrapper,
+};
+use libra_secure_storage::Value;
+use libra_temppath::TempPath;
+use libra_types::waypoint::Waypoint;
+use libra_vm::LibraVM;
+use libradb::LibraDB;
+use storage_interface::DbReaderWriter;
+use structopt::StructOpt;
+
+/// Produces a waypoint from Genesis from the shared storage. It then computes the Waypoint and
+/// optionally inserts it into another storage, typically the validator storage.
+#[derive(Debug, StructOpt)]
+pub struct CreateWaypoint {
+    #[structopt(flatten)]
+    shared_backend: SharedBackend,
+}
+
+impl CreateWaypoint {
+    pub fn execute(self) -> Result<Waypoint, Error> {
+        let genesis_helper = crate::genesis::Genesis {
+            backend: self.shared_backend,
+            path: None,
+        };
+
+        let genesis = genesis_helper.execute()?;
+
+        let path = TempPath::new();
+        let libradb =
+            LibraDB::open(&path, false, None).map_err(|e| Error::UnexpectedError(e.to_string()))?;
+        let db_rw = DbReaderWriter::new(libradb);
+
+        db_bootstrapper::bootstrap_db_if_empty::<LibraVM>(&db_rw, &genesis)
+            .map_err(|e| Error::UnexpectedError(e.to_string()))?
+            .ok_or_else(|| Error::UnexpectedError("Unable to generate a waypoint".to_string()))
+    }
+}
+
+#[derive(Debug, StructOpt)]
+pub struct CreateAndInsertWaypoint {
+    #[structopt(flatten)]
+    shared_backend: SharedBackend,
+    #[structopt(flatten)]
+    validator_backend: ValidatorBackend,
+}
+
+impl CreateAndInsertWaypoint {
+    pub fn execute(self) -> Result<Waypoint, Error> {
+        let waypoint = CreateWaypoint {
+            shared_backend: self.shared_backend,
+        }
+        .execute()?;
+        let storage_name = self.validator_backend.name();
+        let mut validator_storage =
+            StorageWrapper::new(storage_name, &self.validator_backend.validator_backend)?;
+        validator_storage.set(WAYPOINT, Value::String(waypoint.to_string()))?;
+        Ok(waypoint)
+    }
```

Comment:
I would say we could share this between "CreateAndInsert" and "Insert".  But, it's like 4 lines and probably not worth it.

Reply:
hahah that and insert ends up in operational, so we'd have to add it to a library in management. FWIW, we did lose a insert call in config-builder.rs, so it probably makes sense to have a dedicated config-builder crate that can depend on either library.